operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,37 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// driverConfig can be used to specify platform specific driver configuration.
+	// When omitted, this means no opinion and the platform is left to choose reasonable
+	// defaults. These defaults are subject to change over time.
+	// +optional
+	DriverConfig CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
+	// driverName indicates which CSI driver should be installed.
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	DriverName CSIDriverName `json:"driverName"`
```

Comment:
This should be an enum, do we have a list of valid values for this?

Reply:
I have renamed this to `CSIDriverType` and defined a new enum. @bertinatto raised a concern that we were overloading `CSIDriverName` type which is used for defining driver name.